config/config.go

```diff
@@ -0,0 +1,213 @@
+		lc = BombSquadLabelConfig{}
+		lc[string(s.HighCardLabelName)] = encode(mrc)
+	}
+
+	err := WriteBombSquadConfig(b, c)
+	if err != nil {
+		log.Fatalf("Failed to write BombSquadConfig: %s\n", err)
+	}
+}
+
+func DeleteRelabelConfigFromArray(arr []*promcfg.RelabelConfig, index int) []*promcfg.RelabelConfig {
+	res := []*promcfg.RelabelConfig{}
+	if len(arr) > 1 {
+		res = append(arr[:index], arr[index+1:]...)
+	} else {
+		res = []*promcfg.RelabelConfig{}
+	}
+	return res
+}
+
+func FindRelabelConfigInScrapeConfig(encodedRule string, scrapeConfig promcfg.ScrapeConfig) int {
+	for i, relabelConfig := range scrapeConfig.MetricRelabelConfigs {
+		if encode(*relabelConfig) == encodedRule {
+			return i
+		}
+	}
+
+	return -1
+}
+
+func InsertMetricRelabelConfigToPromConfig(rc promcfg.RelabelConfig, c Configurator) promcfg.Config {
+	promConfig := ReadPromConfig(c)
+	rcEncoded := encode(rc)
+	for _, scrapeConfig := range promConfig.ScrapeConfigs {
+		if FindRelabelConfigInScrapeConfig(rcEncoded, *scrapeConfig) == -1 {
+			fmt.Printf("Did not find necessary silence rule in ScrapeConfig %s, adding now\n", scrapeConfig.JobName)
+			scrapeConfig.MetricRelabelConfigs = append(scrapeConfig.MetricRelabelConfigs, &rc)
+		}
+	}
+	return promConfig
+}
+
+func encode(rc promcfg.RelabelConfig) string {
+	b, err := yaml.Marshal(rc)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	s := fmt.Sprintf("%s", string(b))
+	return base64.StdEncoding.EncodeToString([]byte(s))
+}
+
+func ConfigGetRuleFiles() []string {
+	return []string{"nope", "not yet"}
+}
+
+// HighCardSeries represents a Prometheus series that has been idenitified as
+// high cardinality
+type HighCardSeries struct {
+	MetricName        string
+	HighCardLabelName model.LabelName
+}
+
+// TODO: Only generate the relabel config for the appropriate job that is spitting out
+// the high-cardinality metric
+// TODO: Within a job, some series may never be exploding on this label. Consider including
+// all relevant labels in source_labels...?
+func GenerateMetricRelabelConfig(s HighCardSeries) promcfg.RelabelConfig {
+	valueReplace := "bs_silence"
+	regexpOriginal := fmt.Sprintf("^%s;.*$", s.MetricName)
+	promRegex, err := promcfg.NewRegexp(regexpOriginal)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	newMetricRelabelConfig := promcfg.RelabelConfig{
+		SourceLabels: model.LabelNames{"__name__", s.HighCardLabelName},
+		Regex:        promRegex,
+		TargetLabel:  string(s.HighCardLabelName),
+		Replacement:  valueReplace,
+		Action:       "replace",
+	}
+	return newMetricRelabelConfig
+}
+
+func resetMetric(metricName, labelName string) {
+	client, _ := util.HttpClient()
+	// TODO This is a hack currently, to allow the CLI invocation of `unsilence` to actually get to the metric
+	// that needs reset. Should not assume that CLI will be invoked from the running instance, and make this
+	// configurable
+	endpt := fmt.Sprintf("http://localhost:8080/metrics/reset?metric=%s&label=%s", metricName, labelName)
```

Comment:
I agree on the hack. At a bare minimum you should use the flag port here.

Reply:
Agreed, just unaddressed here because there's another issue (#15) for making all of this suck less. :-)